lib/topic_query.rb

```diff
@@ -836,24 +838,17 @@ def default_results(options = {})
     end
 
     if status = options[:status]
-      case status
-      when "open"
-        result = result.where("NOT topics.closed AND NOT topics.archived")
-      when "closed"
-        result = result.where("topics.closed")
-      when "archived"
-        result = result.where("topics.archived")
-      when "listed"
-        result = result.where("topics.visible")
-      when "unlisted"
-        result = result.where("NOT topics.visible")
-      when "deleted"
-        category = Category.find_by(id: options[:category])
-        if @guardian.can_see_deleted_topics?(category)
-          result = result.where("topics.deleted_at IS NOT NULL")
-          require_deleted_clause = false
-        end
-      end
+      options[:q] ||= +""
+      options[:q] << " status:#{status}"
```

Comment:
> What do you think if we collect :q as an array of options or hash of options. 

Can you expand on what you mean here? Currently `q` is a param that represents a custom query language which we intend to support. Therefore, it'll look something like this in the long run: `status:closed category:todo`. We'll need some form or regexp to parse the string one way or another.

Reply:
My understanding was that options are parsed before from query language and we would have something like`options[:status]` and `options[:category]`

If yes, then instead of passing options[:q] to TopicsFilter, maybe we can pass all individual options to TopicFilter
something like:
```
options_hash = {}
options_hash[:status] = options[:status]
options_hash[:category] = options[:category]

 result =
        TopicsFilter.new(
          scope: result,
          guardian: @guardian,
          category_id: options[:category],
        ).filter(options_hash)
```

Not sure if it is possible but if yes, then   TopicsFilter would be a bit clearer what are potential options

Reply:
O ic. I think what you're saying here is to separate out the responsibility of parsing the query language into another class that is different from `TopicsFilter`?